app/models/task.rb

```diff
@@ -994,13 +994,36 @@ def init(task, is_retry)
     end
 
     def make_pdf
-      logger.debug "Running QPDF on all documents before rendering to repair any potential broken files."
+      logger.debug "Rendering PDF for a task, preprocessing attachments."
       @files.each do |f|
         if f[:type] == "document"
+          logger.debug "Running QPDF on #{f[:path]} before rendering to repair any potential broken files."
           FileHelper.qpdf(f[:path])
+        elsif f[:type] == "code"
+          extension = File.extname(f[:path])[1..-1]
+          case extension
+          when "ipynb"
```

Comment:
Why not ipynb? I think these may be more likely to have the embedded data sets.

Reply:
Modifying ipynb without parsing it properly will break the notbook entirely, and jupynotex will fail to process it. The hope is that jupynotex will be able to handle it but I'm not 100% confident that is the case, will have to add another test file to make sure it works properly.

Reply:
Fixed